### DISCUSSION TITLE
Song select design idea (gameplay preview)

### DISCUSSION THREAD

**Discussion by MangoDragon** [ppy/osu#24125] [Ideas]
Hi, I created a slight edit of: https://www.figma.com/file/DXKwqZhD5yyb1igc3mKo1P/Song-Select-Draft-2?node-id=0%3A1 to demonstrate an idea I had. Basically, I thought that instead of having a long rectangular crop of the song's background image in the top left, this could be replaced with a small background image window of the song. If the user presses the play button, a gameplay preview of the song will play in the same window.

I also reduced some of the song information that is shown by default, along with a few minor changes. Extra detail can be displayed by pressing "Show more details".

![song_select_mango_edit](https://github.com/ppy/osu/assets/35911099/0f619748-cac8-4bc6-9da4-47de53e26ed9)

My edit:
https://www.figma.com/file/UCjIQwpQf2WwlI571Fgig1/Song-Select-Draft-2-(Copy)?type=design&node-id=0%3A1&mode=design&t=I20eAWzA1pOkyTGU-1

Of course minor changes can be made, but I think it would be nice to be able to see the whole song background of the song. Please forgive me for misalignment etc, as it was my first time designing with figma.

**Comment by ItsShamed** [comment]
> Basically, I thought that instead of having a long rectangular crop of the song's background image in the top left, this could be replaced with a small background image window of the song. If the user presses the play button, a preview of the song will play in the same window.

This has no purpose as the song is already previewed when you select a song/difficulty. This is just some waste of space. And if you want to listen to the full song while on the song select, you can seek using the music controller.

> I also reduced some of the song information that is shown by default, along with a few minor changes. Extra detail can be displayed by pressing "Show more details".

This strips a lot of useful information, that is the other stats (CS, OD etc.). Also, this feels really empty without a background and will not work well if for example the background is really bright.
And what does the "Show details" button do? Does it redirect to the beatmap listing page? Does it change the layout of the song select?

Finally, the slanting in the leaderboard entries makes it look odd with this kind of straight layout, I don't think it was designed to be laid out like that.

**Reply by MangoDragon** [reply]
The preview would be a gameplay preview. This way you can do things like check the style of the song before you play it. For example, if you are in a multiplayer lobby with a friend, you can check that the song won't be too hard for them before you select it by checking the preview.

My intent was to just quickly display my idea, so some design things might need a bit of work sorry. 🙇‍♂️

**Reply by ItsShamed** [reply]
If anything this would be in a separate section of the "info zone".

Idk if embedding a `Player` inside the song select is viable though. 
I'm particularly scared of the (lazy-)loading flow and performance for long maps (e.g marathon maps), especially since the song select is infamous for being slow and recent work achieved to address this,  would be shameful to lose some recently gained performance because of a QoL feature idk.
